extension/js/common/ui/backup-ui.ts

```diff
@@ -0,0 +1,175 @@
+/* ©️ 2016 - present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com */
+
+'use strict';
+
+import { BackupAutomaticModule } from '../../../chrome/settings/modules/backup-automatic-module.js';
+import { BackupStatusModule } from '../../../chrome/settings/modules/backup-status-module.js';
+import { BackupManualActionModule as BackupManualModule } from '../../../chrome/settings/modules/backup-manual-module.js';
+import { Gmail } from '../api/email-provider/gmail/gmail.js';
+import { View } from '../view.js';
+import { AcctStore, EmailProvider } from '../platform/store/acct-store.js';
+import { BrowserMsg } from '../browser/browser-msg.js';
+import { ClientConfiguration } from '../client-configuration.js';
+import { Xss } from '../platform/xss.js';
+import { KeyIdentity, KeyInfoWithIdentity, KeyUtil } from '../core/crypto/key.js';
+import { KeyStore } from '../platform/store/key-store.js';
+import { Ui } from '../browser/ui.js';
+import { Str } from '../core/common.js';
+import { Lang } from '../lang.js';
+
+export type BackupUiActionType = 'setup_automatic' | 'setup_manual' | 'backup_manual' | undefined;
+export abstract class BackupUi extends View {
```

Comment:
I'm thinking, it would be more flexible maybe for this to be a non-view class? Just something you use inside a view, but not a view itself.

```ts
const backupUi = new BackupUi(...);
await backupUi.initialize();
// or maybe
await backupUi.initialize({
  onBackedUpFinished: () => { ... },
  onXXX // if needed
});
```

Let's say at some point, one view could use more than one of such classes. (I think in TS you cannot inherit from more than one class). I also imagine it would allow for the code to be less interdependent that way - probably a good thing. What do you think?

Reply:
Yeah, I totally agree with you. This way seems more clear.

> await backupUi.initialize({
>   onBackedUpFinished: () => { ... },
>   onXXX // if needed
> });
>